src/index.js

```diff
@@ -289,11 +291,12 @@ export default function undoable (reducer, rawConfig = {}) {
           }
         }
 
-        if (history.present !== res) {
+        // Don't add entry to history if it was just created
+        if ((justCreatedHistory && history.present === res) || action.type === '@@redux-undo/INIT') {
```

Comment:
_(Think you mean `@@redux/INIT`in your first paragraph righ?)_
 
Crap.. good point. 

But if we want to reset to the history upon `@@redux/INIT` to work with HMR then we really gonna have to have that listed right? I think it would be fair to just add both `@@redux/INIT` and `@@redux-undo/INIT` in our `initTypes` config and tell the users that f they remove those types:

* HMR will be broken
* Additional states will be added to the history if those actions are called
* Users could prevent that last side effect by adding a custom filter that filters those actions :)

And on HMR, why don't we create some functional tests with HMR and webpack, haha that would be a nightmare. No, just kidding. Gaearon just wrote a [nice longread](https://medium.com/@dan_abramov/hot-reloading-in-react-1140438583bf#.5ojsji2hb) on the current state of HMR.